Nowsecure/index.ts

```diff
@@ -1,135 +1,149 @@
-// Node code for calling Java app
-//////////////////////////////////////////////////////////////////////////
-import * as tl from "azure-pipelines-task-lib/task";
+import tl = require("azure-pipelines-task-lib/task");
+import tr = require("azure-pipelines-task-lib/toolrunner");
 import fs = require("fs");
 import path = require("path");
 
-let onError = function (errMsg: string, code: number) {
-  tl.error(errMsg);
-  tl.setResult(tl.TaskResult.Failed, errMsg);
+const enum Arch {
+  X86 = "x86",
+  X64 = "amd64",
+  ARM = "arm"
 }
 
-//////////////////////////////////////////////////////////////////////////
-// Read parameters
-//////////////////////////////////////////////////////////////////////////
-let filepath = tl.getInput("filepath", true);
-tl.debug("filepath: " + filepath);
-
-let group = tl.getInput("group", true);
-tl.debug("group: " + group);
-
-let token = tl.getInput("token", true);
-tl.debug("token: " + token);
-
-let username = tl.getInput("username", false);
-tl.debug("username: " + username);
-
-let password = tl.getInput("password", false);
+function platformToString(platform: tl.Platform) {
+  switch (platform) {
+    case tl.Platform.Windows:
+      return "windows";
+    case tl.Platform.Linux:
+      return "linux";
+    case tl.Platform.MacOS:
+      return "darwin";
+  }
+}
 
-let url = tl.getInput("url", false);
-tl.debug("url: " + url);
+function chmodx(toolPath: string, platform: tl.Platform) {
+  if (platform !== tl.Platform.Windows) {
+    tl.execSync("chmod", ["u+x", toolPath])
+  } else {
+    tl.execSync("attrib", ["+x", toolPath])
+  }
+}
 
-let artifactsDir = tl.getInput("artifactsDir", true);
-tl.debug("artifactsDir: " + artifactsDir);
+function archFrom(envvar: string): Arch {
+  switch (envvar) {
+    case "X86":
+      return Arch.X86
+    case "X64":
+      return Arch.X64
+    case "ARM":
+      return Arch.ARM
+    default:
+      const emsg = "Unknown system architecture"
+      tl.setResult(tl.TaskResult.Failed, emsg)
+      throw new Error(emsg)
+  }
+}
 
-let scoreThreshold = tl.getInput("scoreThreshold", false);
-tl.debug("scoreThreshold: " + scoreThreshold);
+function toolName(arch: Arch, platform: tl.Platform): string {
+  // Ex: ns_linux-amd64
+  return `ns_${platformToString(platform)}-${arch}${platform === tl.Platform.Windows ? '.exe' : ''}`
+}
 
-let waitMinutes = tl.getInput("waitMinutes", false);
-tl.debug("waitMinutes: " + waitMinutes);
+function getTool(): tr.ToolRunner {
+  const platform = tl.getPlatform()
+  const arch = archFrom(tl.getVariable("Agent.OSArchitecture"))
+  const toolPath = path.join(__dirname, toolName(arch, platform));
 
-let showStatusMessages = tl.getInput("showStatusMessages", false);
-tl.debug("showStatusMessages: " + showStatusMessages);
+  if (!fs.existsSync(toolPath)) {
+    const err =
+      "Unsupported runner type. Integration currently supports darwin/arm64, windows/amd64, and linux/amd64"
```

Comment:
Eventually we'd want to pull down binaries from the github releases page of that ns repo if they're not present locally